### DISCUSSION TITLE
trigger for "at scheduler startup"

### DISCUSSION THREAD

**Discussion by philsayersCDS** [quartznet/quartznet#980] [General] [answered]
I am using quartznet as part of a larger back office database system. Running reports on schedules, overnight housekeeping tasks...etc. Built around .Net Framework 4.7.2 Currently using QuartzNet 2.x, about to jump over to Quartznet 3.x
I'm using TopShelf and I host Quartznet as a windows service.

Quartznet comes with the SimpleTrigger and CronTrigger "in the box".

I'm using the AdoJobStore, and if possible I would really like to avoid having to use the xml file plugin to make this work.

I need to schedule some jobs to run once at scheduler startup. It looks like this is something I have to build out myself with a custom global listener and some sort of config/settings that I read in that global listener where I define which jobs I want to execute at scheduler startup.

Am I missing something here? As far as I can see, there is no magic CronTrigger pattern or magic string in SimpleTrigger to indicate "RunOnceAtSchedulerStartup".

I see the SchedulerShutdown plugin, but nothing similar for SchedulerStartup.

**Comment by lahma** [comment, marked as answer]
> Quartznet comes with the SimpleTrigger and CronTrigger "in the box".

And there are two more when you have latest version `IDailyTimeIntervalTrigger` and `ICalendarIntervalTrigger` .

I don't think that you need listeners at all. I'd suggest that when you build your main scheduler just call:

`IScheduler.TriggerJob(JobKey jobKey)`

This will create a run-once simple trigger under the hood and run it as soon as possible.

**Reply by philsayersCDS** [reply]
I have about 50  deployments and each one need zero, one or several different jobs executing at scheduler startup.

I was hoping there was a way to  accomplish this where the trigger would be stored in the job store forever.  So I  (or my customers) could use something like quartzmin to setup the trigger once, the trigger would get stored in the adojobstore and that would be it.  I wouldn't need to have anything unique to a particular customer baked into my code.

I can cobble something together with an xml file for "scheduler startup jobs" and follow the spirit of the standard xml plugin to keep this flexible.

**Reply by lahma** [reply]
That might sound like something more like a migration framework that would add flags to database whether something has run or not, and then schedule on startup and check if there's a need to run and if there is, do the job and update the flag.
